scripts/emerge/ingest_to_tdr.py

```diff
@@ -21,84 +21,116 @@ def get_access_token():
     """Create the ingestDataset API json request body and call API."""
 
-    load_json = json.dumps({"format": "json",
-                            "path": control_file_path,
-                            "table": target_table_name,
-                            "resolve_existing_files": "true",
-                            "updateStrategy": "replace"
-                            })
-    load_job_response = load_data(dataset_id, load_json)
-
-    print(f"ingestDataset request body: \n {load_json} \n")
-    print(f"ingestDataset response body: \n {load_job_response} \n")
-
-    ingestDataset_job_id = load_job_response["id"]
-    ingestDataset_status_code = load_job_response["status_code"]
-
-    if ingestDataset_status_code != 202:
-        print("file ingest to TDR dataset failed.")
-        print(f"please refer to error message and retry ingest.")
-        print(f"ingestDataset response body: \n {load_job_response} \n")
-        raise ValueError(load_job_response)
-
-    # if ingestDataset_status_code == 202:  # if ingestDataset starts running successfully
-    job_status_code, job_response = get_job_status_and_result(ingestDataset_job_id)
-    while job_status_code == 202:           # while job is still running
-        time.sleep(10)  # wait 10 seconds
-        job_status_code, job_response = get_job_status_and_result(ingestDataset_job_id)
-
-    if job_status_code != 200:              # when job completes but not success
-        error_message = job_response["errorDetail"]
-        print(f"Load job finished but did not succeed: {error_message}")
-        raise ValueError(error_message)
+    ingest_dataset_request = create_ingest_dataset_request(control_file_path, target_table_name, load_tag) # create request for ingestDataset
+    print(f"ingestDataset request body: \n {ingest_dataset_request} \n")
+
+    ingest_response = ingest_dataset(dataset_id, ingest_dataset_request) # call ingestDataset
+    print(f"ingestDataset response body: \n {ingest_response} \n")
+
+    ingest_job_id = ingest_response["id"] # check for presence of id in ingest_dataset()
+    ingest_status_code = ingest_response["status_code"]
+
+    if ingest_status_code != 202: # job submitted but fails
+        print(f"{ingest_job_id} --> failed")
+        raise ValueError(ingest_response)
+
+    # 202 = job is running
+    job_status_code, job_status_response = get_job_status(ingest_job_id)
+
+    while job_status_code == 202:           # while job is running
+        print(f"{ingest_job_id} --> running")
+        time.sleep(10) # wait 10 seconds before getting job status
+        job_status_code, job_status_response = get_job_status(ingest_job_id) # get updated status info
+
+    # job completes (‘failed’ or ‘succeeded’) with 200 status_code
+    # consider any combination other than 200 + succeeded as failure
+    if job_status_code != 200 or job_status_response["job_status"] != "succeeded":
+        print(f"{ingest_job_id} --> failed")
```

Comment:
hmm if job_status_code is not 200, if it's e.g. 500 (something wrong with the service transiently) that doesn't necessarily mean the job failed, right? this might be enough of an edge case not to worry about but i think the logic here isn't quite water tight

Reply:
yeah i wanted to handle it "simple" in that if its not a true true success - aka the 200 and a "success" then its considered as a fail so as not to run the risk of failed ingests into TDR. Aka re-run it if it fails for any reason other than fully successful. sort of naive but i think operating in a "better safe than sorry" method here for emerge since its a larger scale number of ingests.

Reply:
that makes sense - but in that case then i'm not sure that what you wrote in row 126 is guaranteed to be true

if you want it to be true then i'd change row 119 to
`if not (job_status_code == 200 and job_status_response["job_status"] == "succeeded"):`

Reply:
smart using the `not`

i was playing around with
```
    if job_status_code == 200 and job_status_response["job_status"] == "succeeded":
        # job completes successfully
        # success_code is 200 and "job_status" =is"succeeded"
        print(f"{ingest_job_id} --> succeeded")

print(f"{ingest_job_id} --> failed")
# if failed, get the resulting error message
job_result_code, job_result_response = get_job_result(ingest_job_id)
raise ValueError(job_result_response)   
``` 
but i didnt want the flow of conditionals to flip around from what it was above.

thanks for the suggestions it sorts out what i was trying to get to